SumoLogic.Logging.Common/Sender/SumoLogicMessageSender.cs

```diff
@@ -259,50 +256,41 @@ public async Task TrySend(string body, string name, string category, string host
                         this.Log.Debug("Successfully sent log request to Sumo Logic");
                     }
                 }
-                catch (AggregateException ex)
+                catch (OperationCanceledException ex)
                 {
                     if (this.Log.IsWarnEnabled)
                     {
-                        this.Log.Warn("Could not send log to Sumo Logic");
+                        this.Log.Warn("Could not send log to Sumo Logic. Operation was canceled");
                     }
-
-                    if (this.Log.IsDebugEnabled)
+                    else if (this.Log.IsDebugEnabled)
                     {
-                        this.Log.Debug("Reason: " + ex.InnerException.Message);
+                        this.Log.Debug($"Could not send log to Sumo Logic. {ex.GetType()}: {ex.Message}");
                     }
-
-                    throw ex.InnerException;
+                    // No rethrow, the operation cannot be retried

```

Comment:
`OperationCanceledException´ usually means HttpClient has been closed/disposed. This usually means that we want to close/shutdown. This is usually not a time to perform retries, but a time where one should get out of the burning house.

Reply:
I understand that if HttpClient is disposed, we would not like to retry, but won't `ObjectDisposedException` be thrown in that case (I didn't add a comment in that catch block :) )?
You are catching both exceptions, I am assuming:
* OperationCanceledException -> for HttpClient couldn't reach the destination.
* ObjectDisposedException -> for HttpClient has been disposed.

Is there anything which I am missing?

Reply:
No you are right. HttpClient can throw OperationCanceledException when Http-connectivity issues. Just used to this exception only being thrown by the task-library (when CancellationToken is set). Will change so it will retry, unless about to close.

Reply:
Fixed